irr/include/IEventReceiver.h

```diff
@@ -543,8 +543,9 @@ struct SEvent
 	};
 
 	SEvent() {
-		// would be left uninitialized in many places otherwise
-		MouseInput.Simulated = false;
+		EventType = static_cast<EEVENT_TYPE>(0);
+		// zero the biggest union member we have, which clears all others too
+		memset(&JoystickEvent, 0, sizeof(JoystickEvent));
```

Comment:
Just morbidly curious; is there a C++ way to deduce this? That would be clever.

Reply:
with c++26 reflection: maybe
otherwise: no

Reply:
> c++26

Calm down son, we are still barely hanging onto C++17